src/BloomExe/Extensions.cs

```diff
@@ -11,19 +11,44 @@ public static string ToLocalhost(this string fileName)
 			// don't do this if it is done already
 			if (fileName.StartsWith(ServerBase.PathEndingInSlash)) return fileName;
 
-			// BL-117, PH: With the newer xulrunner, javascript code with parenthesis in the URL is not working correctly.
-			fileName = fileName.Replace("(", "%28").Replace(")", "%29");
-
-			return ServerBase.PathEndingInSlash + fileName.Replace(":", "%3A").Replace("\\", "/");
+			return ServerBase.PathEndingInSlash + fileName.EscapeCharsForHttp().Replace(System.IO.Path.DirectorySeparatorChar, '/');
 		}
 
 		public static string FromLocalhost(this string uri)
 		{
 			if (uri.StartsWith(ServerBase.PathEndingInSlash))
-				uri = uri.Substring(ServerBase.PathEndingInSlash.Length).Replace("%3A", ":");
+				uri = uri.Substring(ServerBase.PathEndingInSlash.Length).UnescapeCharsForHttp();
 			return uri;
 		}
 
+		/// <summary>
+		/// Escapes a number of characters that need it for url/http processing.  A much larger
+		/// number could be handled, but the general case (escape all non-ASCII chars as well
+		/// as several more ASCII chars) is much more complicated, and the following appears
+		/// to suffice for our needs in communicating to our own localhost processor.
+		/// </summary>
```

Comment:
Because we escape and unescape % itself, we must Unescape exactly as many times as a string has been escaped...unlike To/FromLocalhost, it won't work to Escape something twice and expect to have the original back by unescaping it once, or to just Escape something because you're not sure whether it's already been done.
These are new routines, so that isn't a critical problem, but it might be worth mentioning in a comment.